lib/teiserver/account/libs/relationship_lib.ex

```diff
@@ -388,4 +388,25 @@ defmodule Teiserver.Account.RelationshipLib do
 
     results.rows
   end
+
+  # Deletes inactive users in the ignore/avoid/block list of a user
+  # Returns the number of deletions
+  def delete_inactive_ignores_avoids_blocks(user_id, days_not_logged_in) do
+    query = """
+    delete from account_relationships ar
+      using account_users au
+      where au.id = ar.to_user_id
+      and ar.from_user_id = $1
+      and (au.last_login is null OR
+      abs(DATE_PART('day', (now()- au.last_login ))) > $2);
+    """
+
+    results =
+      Ecto.Adapters.SQL.query!(Repo, query, [
+        user_id,
+        days_not_logged_in
+      ])
+
+    results.num_rows
```

Comment:
Any sample code for this? I have no idea how to do this.

Reply:
no need for a friend request, you can directly use [create_friend](https://github.com/geekingfrog/teiserver/blob/d7df6d3d168415d4932b4b6605063743e48be969/lib/teiserver/account.ex#L1680-L1687) since you only care about the db relationship, not the flow leading to create this relationship.

And to create the user, that's the idea. I suggest you locate/create a function like `user_attrs` that will return the plain map and put that in the fixture folder in tests. This way, you can create a default map, modify it for the purpose of you tests, and then call `create_user` on it. I'm using this pattern a bit for testing oauth stuff: https://github.com/geekingfrog/teiserver/blob/d7df6d3d168415d4932b4b6605063743e48be969/test/teiserver/o_auth/gen_token_test.exs#L13